rpc/src/v1/helpers/engine_signer.rs

```
@@ -36,16 +36,14 @@ impl EngineSigner {
 
 impl engine::signer::EngineSigner for EngineSigner {
 	fn sign(&self, message: Message) -> Result<Signature, Error> {
-		match self.accounts.sign(self.address, Some(self.password.clone()), message) {
-			Ok(ok) => Ok(ok),
-			Err(_) => Err(Error::InvalidSecretKey),
-		}
+		self.accounts.sign(self.address, Some(self.password.clone()), message).map_err(|e| {
+			Error::Custom(e.to_string())
```

propagate real error cause instead of making assumptions.